a3p-integration/proposals/n:upgrade-next/package.json

```
@@ -8,16 +8,9 @@
     },
     "type": "Software Upgrade Proposal",
     "sdk-generate": [
-      "testing/replace-feeDistributor-short.js replaceFeeDistributor",
       "testing/add-USD-LEMONS.js addUsdLemons",
-      "vats/upgrade-provisionPool.js upgradeProvisionPool",
-      "vats/upgrade-asset-reserve.js upgradeAssetReserve",
-      "vats/upgrade-psm.js upgradePSM",
-      "vats/upgrade-paRegistry.js",
-      "vats/upgrade-agoricNames.js agoricNamesCoreEvals/upgradeAgoricNames",
       "testing/add-USD-OLIVES.js agoricNamesCoreEvals/addUsdOlives",
-      "testing/publish-test-info.js agoricNamesCoreEvals/publishTestInfo",
-      "vats/upgrade-mintHolder.js upgrade-mintHolder A3P_INTEGRATION"
+      "testing/publish-test-info.js agoricNamesCoreEvals/publishTestInfo"
```

all these proposals for testing should be under `test/` so they don't invalidate the "use" image. see,
- https://github.com/Agoric/agoric-3-proposals/pull/213

Making that change affects .gitignore and .tsconfig but if you put them under a path like `test/submissions` then you can use that pathname for those files and not have to manage a list of submissions.

We'll make this easier eventually,
- https://github.com/Agoric/agoric-sdk/issues/10537

---

Actually they should be in `host`, and their core eval output should be in `test`

Something like 

```suggestion
      "host/testing/publish-test-info.js test/agoricNamesCoreEvals/publishTestInfo"
```

---

FYI @Chris-Hibbert , the `host` thing hasn't landed in a3p. Up to you whether to wait for it to be merged and released to NPM so you can include that synthetic-chain lib version in this PR.

- https://github.com/Agoric/agoric-3-proposals/pull/210/

---

I like your `test/generated` approach. It still copies to the image unnecessarily, but it's only the test image so it doesn't invalid any extra. (If the proposal changes its build would anyway). I think it's better to keep test stuff together than to spread it into another path.

---

Yeah you're right, a change in the proposal building would likely invalidate everything anyway, but collocating test files together is cleaner.

I got confused about the `host` stuff however, I forgot these first argument is relative to the SDK packages.

---

I moved the tests to `test/`. I was unable to move the test library under test (lint interpreted it as including test code, which is forbidden), so it remains at top-level as `n:upgrade-next/test-lib`. The `host` stuff seems to be related to `agoric-3-proposals`, not script building proposals in `agoric-sdk`.

---

> The `host` stuff seems to be related to `agoric-3-proposals`, not script building proposals in `agoric-sdk`.

Yes sorry I got confused with the `host` stuff.

This looks great now